extension.ts

```diff
@@ -0,0 +1,117 @@
+    await vscode.window.showErrorMessage(error.message)
+}
+
+const handleCommandErrors = <P extends unknown[], R>(command: (...args: P) => Promise<R>) => async (
+    ...args: P
+): Promise<R | void> => {
+    try {
+        return await command(...args)
+    } catch (error) {
+        await showError(error)
+    }
+}
+
+/**
+ * The command implementation for opening a cursor selection on Sourcegraph.
+ */
+async function openCommand(): Promise<void> {
+    const editor = vscode.window.activeTextEditor
+    if (!editor) {
+        throw new Error('No active editor')
+    }
+    const repositoryInfo = await repoInfo(editor.document.uri.fsPath)
+    if (!repositoryInfo) {
+        return
+    }
+    const { remoteURL, branch, fileRelative } = repositoryInfo
+
+    // Open in browser.
+    await open(
+        `${getSourcegraphUrl()}/-/editor` +
+            `?remote_url=${encodeURIComponent(remoteURL)}` +
+            `&branch=${encodeURIComponent(branch)}` +
+            `&file=${encodeURIComponent(fileRelative)}` +
+            `&editor=${encodeURIComponent('VSCode')}` +
+            `&version=${encodeURIComponent(version)}` +
+            `&start_row=${encodeURIComponent(String(editor.selection.start.line))}` +
+            `&start_col=${encodeURIComponent(String(editor.selection.start.character))}` +
+            `&end_row=${encodeURIComponent(String(editor.selection.end.line))}` +
+            `&end_col=${encodeURIComponent(String(editor.selection.end.character))}`
+    )
+}
+
+/**
+ * The command implementation for searching a cursor selection on Sourcegraph.
+ */
+async function searchCommand(): Promise<void> {
+    const editor = vscode.window.activeTextEditor
+    if (!editor) {
+        throw new Error('No active editor')
+    }
+    const repositoryInfo = await repoInfo(editor.document.uri.fsPath)
+    if (!repositoryInfo) {
+        return
+    }
+    const { remoteURL, branch, fileRelative } = repositoryInfo
+
+    const query = editor.document.getText(editor.selection)
+    if (query === '') {
+        return // nothing to query
+    }
+
+    // Search in browser.
+    await open(
+        `${getSourcegraphUrl()}/-/editor` +
+            `?remote_url=${encodeURIComponent(remoteURL)}` +
+            `&branch=${encodeURIComponent(branch)}` +
+            `&file=${encodeURIComponent(fileRelative)}` +
+            `&editor=${encodeURIComponent('VSCode')}` +
+            `&version=${encodeURIComponent(version)}` +
+            `&search=${encodeURIComponent(query)}`
+    )
+}
+
+/**
+ * The command implementation for searching on Sourcegraph.com
+ */
+
+ async function ossSearchCommand(value: string|undefined): Promise<void> {
+		if(!value){
+			return;
+		}
+		 await open(
+			`https://sourcegraph.com/search?patternType=literal` +
+				`&q=${encodeURIComponent(value)}`
+		);
+}
+
+vscode.commands.registerCommand('extension.ossSearch', () => {
+	vscode.window.showInputBox().then(
+		ossSearchCommand
+	);
+});
```

Comment:
Can we register this command the same way that we register our existing commands?
- register on activation in the `activate` function, add to `context.subscriptions` (read [this](https://stackoverflow.com/a/55579618) for more context on why this matters. we have [the same concept](https://docs.sourcegraph.com/extensions/authoring/activation#deactivation) for Sourcegraph extensions)
- wrap the command function with `handleCommandErrors`